mantidproject/mantid
# Indirect Energy Transfer - Incorrect histories causing large file sizes

**Description of work.**
This PR fixes a problem where the histories of all the workspaces being reduced on the Indirect Data Reduction Energy Transfer interface appears on all of the output reduced files.
This causes a reduced file to have a variable size depending on how many other files were reduced in the same batch. These large output file sizes will begin to cause problems when reduced hundreds of run numbers at a time.

I believe this unwanted history is caused by the `GroupWorkspaces` algorithm which appears to merge the histories of each of the workspaces being grouped. The solution decided upon was to create a Wrapper algorithm so that the history could be turned off for the `ISISIndirectEnergyTransfer` algorithm and so that it will still work with project recovery even though this history is turned off.

**To test:**
1.`Interfaces`->`Indirect`->`Data Reduction`->`Energy Transfer` interface
2. Make sure the default facility is ISIS
3. Instrument is `IRIS`, Analyser is `Graphite` and Reflection is `002`
4. Type in run numbers 83467-83472
5. Click Run and save the first (`iris83467_graphite002_red`) reduced workspace
6. Do more runs with run numbers 83467-83477 and then 83467-83489 saving the output `iris83467_graphite002_red` workspace each time.
7. Check that the size of these files are roughly the same
8. Check that the history of these files contains `ISISIndirectEnergyTransferWrapper` which should have no child algorithms in its history

The file sizes from before the fix and after the fix when run by me:
![image](https://user-images.githubusercontent.com/40830825/53476753-0ddcda00-3a6b-11e9-8473-240fe117a9f3.png)

**A:**
To test that the EnergyTransfer interface produces the same workspace as before (identical data):
1. Enter a run number of `26176`
2. Click `Run` and wait
3. Compare the outputted workspace to: [iris26176_graphite002_red_before.zip](https://github.com/mantidproject/mantid/files/2914586/iris26176_graphite002_red_before.zip)
4. Using CompareWorkspaces, the output table should be empty

**B:**
To test that the Calibration interface produces the same resolution workspace as before (identical data):
1. Enter a run number of `26176`
2. Tick `Create RES File`
3. Click `Run` and wait
3. Compare the outputted _res workspace to: [irs26176_graphite002_res_before.zip](https://github.com/mantidproject/mantid/files/2914582/irs26176_graphite002_res_before.zip)
4. Using CompareWorkspaces, the output table should be empty

Fixes #24582

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
